subsys/fs/ext2/Kconfig

```diff
@@ -0,0 +1,42 @@
+# Copyright (c) 2023 Antmicro
+# SPDX-License-Identifier: Apache-2.0
+
+# Ext2 file system
+
+config FILE_SYSTEM_EXT2
+	bool "Ext2 file system support"
+	depends on FILE_SYSTEM
+	help
+	  Enable Ext2 file system support.
+
+module = EXT2
+module-str = Ext2
+source "subsys/logging/Kconfig.template.log_config"
+
+if FILE_SYSTEM_EXT2
+
+menu "Ext2 file system Settings"
+	visible if FILE_SYSTEM_EXT2
+
+config EXT2_HEAP_SIZE
+	int "Heap size"
+	default 10240
+	help
+	  Memory for allocation of internal structures (e.g. inodes).
+
+config MAX_INODES
+	int "Maximum number of opened inodes"
+	default 10
+
+config EXT2_BLOCKS_BUF_SIZE
+	int "Maximum size of pages buffer in blocks cache."
+	default 40960
+	help
+	  Maximum size of buffer that holds memory for cached pages.
```

Comment:
For me it was more obvious to do it like that, because the size of block may differ for different file systems mounted. I wanted to make an upper bound on the memory that is used to store blocks memory. I was choosing between these two options:

a) Specify the size of buffer that will hold blocks memory
b) Specify the number of blocks and size of them and calculate size of buffer

If you think that the second approach might be better for some reason I will change it. Otherwise I will write a more descriptive help message to better understand what this flag is doing.

Reply:
> For me it was more obvious to do it like that, because the size of block may differ for different file systems mounted. I wanted to make an upper bound on the memory that is used to store blocks memory. I was choosing between these two options:
> 
> a) Specify the size of buffer that will hold blocks memory b) Specify the number of blocks and size of them and calculate size of buffer
> 
> If you think that the second approach might be better for some reason I will change it. Otherwise I will write a more descriptive help message to better understand what this flag is doing.

b) is completely acceptable, that is what actually FAT does.
You can add Kconfig that will state the MAX allowed block size to be mounted, FAT does that with Kconfig FS_FATFS_MAX_SS and then allowed block count. The max allowed block size is quite obvious parameter, but how number of blocks impacts operation. FAT FS is also able to state minimal size of block.
You want to make these settings clear to make it less magic, otherwise people will try to tune these down by trial and error or by random manner, and will be frustrated that sometimes their system does not work or starts failing.

Reply:
Okay, I have fixed it and now we have flags to specify maximum size of block and maximum number of blocks.